File: utils/bot.js
// server.js
const express = require('express');
const { Client, GatewayIntentBits, EmbedBuilder } = require('discord.js');
const dotenv = require('dotenv');
const fs = require('fs');
const path = require('path');

// Load variables from the .env file
dotenv.config({ path: '../mineflared.env' });

// Discord bot configuration
const botToken = process.env['DISCORD-BOT-TOKTEN'];
const serverId = process.env['DISCORD-SERVER-ID'];
const chatId = process.env['DISCORD-CHAT-ID'];

// Initialize Discord client
const client = new Client({
  intents: [GatewayIntentBits.Guilds, GatewayIntentBits.GuildMessages, GatewayIntentBits.MessageContent],
});

client.once('ready', () => {
  console.log('Bot connected to Discord!');
});

// Create an Express server
const app = express();
app.use(express.json());

// Endpoint to send the .md file content as an embed
app.post('/send', async (req, res) => {
  try {
    // Ensure the bot is connected
    if (!client.isReady()) {
      await client.login(botToken);
    }

    // Extract the .md file path from the request body
    const { filePath } = req.body;

    // Check if filePath is provided and valid
    if (!filePath || !fs.existsSync(filePath) || path.extname(filePath) !== '.md') {
      return res.status(400).json({ message: 'Invalid .md file path' });
    }

    // Read the file content
    const fileContent = fs.readFileSync(filePath, 'utf-8');

    // Create an embed message for Discord
    const embed = new EmbedBuilder()
      .setTitle('Markdown File Content')
      .setDescription('The content of the Markdown file is below:')
      .setColor(0x00ff00)
      .addFields({
        name: 'File Content',
        value: `\`\`\`md\n${fileContent}\n\`\`\``, // Enclose the markdown content in code block
      })
      .setTimestamp();

    // Send the embed to the Discord channel
    const channel = await client.channels.fetch(chatId);
    if (channel) {
      await channel.send({ embeds: [embed] });
      res.status(200).json({ message: 'Message sent successfully!' });
    } else {
      res.status(404).json({ message: 'Channel not found!' });
    }
  } catch (error) {
    console.error('Error sending message:', error);
    res.status(500).json({ message: 'Internal error while sending message' });
  }
});

// Start the server on port 8080
const port = 8080;
app.listen(port, () => {
  console.log(`Server running at http://localhost:${port}`);
});

File: utils/bot.py
import discord
from flask import Flask, request, jsonify
from flask_cors import CORS
import os
import asyncio
import json
import threading

# Load environment variables from env.json file
def load_env_from_json(filepath):
    with open(filepath, 'r') as f:
        return json.load(f)

env_vars = load_env_from_json('../env.json')

# Define intents
intents = discord.Intents.default()
intents.messages = True
intents.guilds = True

# Create the Discord bot with intents
bot = discord.Client(intents=intents)

# Get environment variables
server_id = env_vars.get('DISCORD_SERVER_ID')
chat_id = env_vars.get('DISCORD_CHAT_ID')
bot_token = env_vars.get('DISCORD_BOT_TOKEN')

if server_id is None:
    raise ValueError("Key 'DISCORD_SERVER_ID' not found in env.json.")
if chat_id is None:
    raise ValueError("Key 'DISCORD_CHAT_ID' not found in env.json.")
if bot_token is None:
    raise ValueError("Key 'DISCORD_BOT_TOKEN' not found in env.json.")

# Ensure IDs are integers
server_id = int(server_id)
chat_id = int(chat_id)

@bot.event
async def on_ready():
    print(f'Bot connected to Discord server with ID {server_id}.')

# Create the Flask server
app = Flask(__name__)
CORS(app, resources={r"/send": {"origins": "http://127.0.0.1"}})

# Endpoint to send a message
@app.route('/send', methods=['POST'])
def send_message():
    try:
        data = request.get_json()
        message = data.get('message', '')

        if not message:
            return jsonify({'message': 'No message provided'}), 400

        async def send_discord_message():
            channel = bot.get_channel(chat_id)
            if channel is None:
                return jsonify({'message': 'Channel not found!'}), 404

            embed = discord.Embed(
                title="Minecraft Server Link",
                description=f"Server info:\n\n{message}",
                color=discord.Color.green()
            )
            embed.timestamp = discord.utils.utcnow()
            await channel.send(embed=embed)

        # Schedule the message to be sent asynchronously
        asyncio.run_coroutine_threadsafe(send_discord_message(), asyncio.get_event_loop())

        return jsonify({'message': 'Message sent successfully!'}), 200

    except Exception as error:
        print(f"Error sending message: {error}")
        return jsonify({'message': 'Internal error while sending message'}), 500

# Function to run Flask in a separate thread
def run_flask():
    app.run(port=8080)

# Start both the bot and the Flask server
if __name__ == '__main__':
    # Run Flask in a separate thread
    flask_thread = threading.Thread(target=run_flask)
    flask_thread.start()

    # Run the bot on the main thread
    loop = asyncio.get_event_loop()
    loop.run_until_complete(bot.start(bot_token))
